source/PlayerInfo.cpp

```diff
@@ -2142,6 +2165,61 @@ map<string, string> PlayerInfo::GetSubstitutions() const
 
 
 
+bool PlayerInfo::SetTribute(const Planet *planet, int64_t payment)
+{
+	if(payment > 0)
+	{
+		tributeReceived[planet] = payment;
+		// Properly connect this function to the dominated property of planets.
+		GameData::GetPolitics().DominatePlanet(planet);
+	}
+	else
+	{
+		tributeReceived.erase(planet);
+		// Properly connect this function to the (no longer) dominated property of planets.
+		GameData::GetPolitics().DominatePlanet(planet, false);
+	}
+
+	return true;
+}
```

Comment:
It returns a bool for the sake of the prefixed condition provider, and is always true because you can never fail to set your tribute. Although perhaps the condition provider could just return true on its own without relying on this function's return type.

Reply:
Leaving this as is, because the other SetTribute function also returns a bool and can return either true or false.